src/serializer.rs

```diff
@@ -53,325 +45,48 @@ impl<W> Serializer<W>
-                     pub fn path(&self) -> &::std::path::Path {
-                         match *self {
-                             DirEntry::Dir(d) => d.path(),
-                             DirEntry::File(f) => f.path(),
-                         }
-                     }
-                 }")?;
-
-        Ok(self)
-    }
-
-    fn write_dirwalker_definition(&mut self) -> Result<&mut Self> {
-        writeln!(self.writer,
-                 "{}",
-                 "/// A directory walker.
-                  ///
-                  /// `DirWalker` is an iterator which will recursively traverse
-                  /// the embedded directory, allowing you to inspect each item.
-                  /// It is largely modelled on the API used by the `walkdir`
-                  /// crate.
-                  ///
-                  /// You probably won't create one of these directly, instead
-                  /// prefer to use the `Dir::walk()` method.")?;
-
-        writeln!(self.writer, "#[derive(Debug, PartialEq, Clone)]")?;
-        writeln!(self.writer,
-                 "{}",
-                 "pub struct DirWalker<'a> {
-                     root: &'a Dir,
-                     entries_to_visit: ::std::collections::VecDeque<DirEntry<'a>>,
-                 }")?;
-
-        writeln!(self.writer,
-                 "{}",
-                 "impl<'a> DirWalker<'a> {
-                     fn new(root: &'a Dir) -> DirWalker<'a> {
-                         let mut walker = DirWalker{
-                            root: root,
-                            entries_to_visit: ::std::collections::VecDeque::new(),
-                         };
-                         walker.extend_contents(root);
-                         walker
-                     }
-
-                     fn extend_contents(&mut self, from: &Dir) {
-                         for file in from.files {
-                             self.entries_to_visit.push_back(DirEntry::File(file));
-                         }
-
-                         for dir in from.subdirs {
-                             self.entries_to_visit.push_back(DirEntry::Dir(dir));
-                         }
-                     }
-                 }")?;
-
-        writeln!(self.writer,
-                 "{}",
-                 "impl<'a> Iterator for DirWalker<'a> {
-                    type Item = DirEntry<'a>;
-
-                    fn next(&mut self) -> Option<Self::Item> {
-                        let entry = self.entries_to_visit.pop_front();
-
-                        if let Some(DirEntry::Dir(d)) = entry {
-                            self.extend_contents(d);
-                            Some(DirEntry::Dir(d))
-                        } else {
-                            entry
-                        }
-                    }
-                }")?;
-
+    fn write_std_definitions(&mut self) -> Result<&mut Self> {
+        self.writer.write(include_bytes!("serialized_std_definitions.rs"))?;
```

Comment:
Good idea removing the string literals and replacing them with `include_bytes!()` :+1:

Reply:
The output formatting is now correct as well ☮️